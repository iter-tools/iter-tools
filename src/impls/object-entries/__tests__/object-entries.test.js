import { objectEntries } from 'iter-tools-es';

describe('objectEntries', () => {
  it('works with Objects', () => {
    const i = objectEntries({ 1: 1, 2: 2 });
    expect(Array.from(i)).toEqual([
      ['1', 1],
      ['2', 2],
    ]);
  });

  it('can be reused', () => {
    const i = objectEntries({ 1: 1, 2: 2 });
    expect(Array.from(i)).toEqual([
      ['1', 1],
      ['2', 2],
    ]);
    expect(Array.from(i)).toEqual([
      ['1', 1],
      ['2', 2],
    ]);
  });

  it('works with null', () => {
    const i = objectEntries(null);
    expect(Array.from(i)).toEqual([]);
  });

  it('throws with non-object', () => {
    const obj: any = 4;
    expect(() => objectEntries(obj)).toThrowErrorMatchingSnapshot();
  });
});
