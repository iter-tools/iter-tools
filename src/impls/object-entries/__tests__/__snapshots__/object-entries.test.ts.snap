// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`objectEntries throws with non-object 1`] = `"obj argument to objectEntries must be an object, null, or undefined"`;
